storage/libradb/src/ledger_store/mod.rs

```diff
@@ -33,11 +34,31 @@ use types::{
 
 pub(crate) struct LedgerStore {
     db: Arc<DB>,
+
+    /// We almost always need the latest ledger info and signatures to serve read requests, so we
+    /// cache it in memory in order to avoid reading DB and deserializing the object frequently. It
+    /// should be updated every time new ledger info and signatures are persisted.
+    latest_ledger_info: ArcSwap<Option<LedgerInfoWithSignatures<Ed25519Signature>>>,
```

Comment:
Is this a good use case for ArcSwapOption?  The nesting is reversed ( Option<ArcSwap> ) but since it should just be a pointer swap, I think it might function the same.

Reply:
Yeah, they are similar. Based on the existing code it seems like `ArcSwap<Option>` is slightly more convenient.